startup/server.js

```diff
@@ -0,0 +1,78 @@
+/**
+ * @fileoverview The server is in charge of serving webpages, managing Lobby objects,
+ * and handling/passing websocket data to the correct lobby.
+ */
+const http = require('http');
+const ws = require('ws');
+
+
+/**
+ * SERVER_PORT is the port the NodeJS server should constantly listen to for incoming websocket requests. Port 80 is for listening to client HTTP requests.
+ * Client-side JS from port 80 (connected through a regular HTTP request that can be made from the browser) makes a connection to Port 8000 for the websocket.
+ */
+const SERVER_PORT = 8000; 
+var s = new Server();
+/**
+ * @constructor
+ */
+ function Server() {
+    this.webSocketServer = new ws.Server( { noServer: true });
+    this.lobbies = {}; // A dictionary of lobbies indexed by their id
+}
+
+/**
+ * To be called when a websocket connection is made to the websocket server
+ * @param {WebSocket} socket The new socket connection
+ */
+Server.prototype.onOpen = function(socket) {
```

Comment:
Same here. Do these codes work? Unclear if this actually takes a socket as an input. Also, we on the server side decided not to make the server a class because it would be easier to work with.

Reply:
The socket will need to be a parameter of some sort in order for it to work. Additional ones might be needed in the future depending on backend needs